lib/snippet-expansion.coffee

```diff
@@ -10,17 +10,32 @@ class SnippetExpansion
     @selections = [@cursor.selection]
 
     startPosition = @cursor.selection.getBufferRange().start
+    # Get leading indentation level
+    indent = @editor.lineTextForBufferRow(startPosition.row).match(/^\s*/)[0]
+    # Add proper indentation to the snippet
+    body = @snippet.body.replace(/\n/g, '\n' + indent)
+
+    tabStopRanges = []
```

Comment:
```diff
diff --git a/lib/snippet-expansion.coffee b/lib/snippet-expansion.coffee
index bec664d..8902e51 100644
--- a/lib/snippet-expansion.coffee
+++ b/lib/snippet-expansion.coffee
@@ -15,17 +15,14 @@ class SnippetExpansion
     # Add proper indentation to the snippet
     body = @snippet.body.replace(/\n/g, '\n' + indent)

-    tabStopRanges = []
-    for tabStop in @snippet.tabStops
-      ranges = []
-      for range in tabStop
+    tabStopRanges = @snippet.tabStops.map (tabStop) ->
+      tabStop.map (range) ->
         newRange = Range.fromObject(range, true) # Don't overwrite the existing range
         if newRange.start.row # a non-zero start row means that we're not on the initial line
           # Add on the indent offset so that the tab stops are placed at the correct position
           newRange.start.column += indent.length
           newRange.end.column += indent.length
-        ranges.push(newRange)
-      tabStopRanges.push(ranges)
+        newRange

     @editor.transact =>
       newRange = @editor.transact =>
```